src/main/java/org/reactome/qa/doisuggester/Main.java

```diff
@@ -50,213 +56,201 @@ public static void main(String[] args) throws FileNotFoundException, IOException
 
-					// Iterate through all new instance edits, trying to find a Pathway containing
-					// them.
-					for (GKInstance rleIE : newRLEInstanceEdits)
+			if (!newInstanceEdits.isEmpty())
+			{
+				// Now it will check the immediate parent Pathway of the RlE to see if the new
+				// IE exists there.
+				// If it is not found, it will continue to check up the Pathway hierarchy to see
+				// if a new IE exists anywhere.
+
+				// Note: Expand so that instances with multiple parents are handled.
+				GKInstance parentPathway = (GKInstance) currentRLE.getReferers(ReactomeJavaConstants.hasEvent).iterator().next();
+				List<GKInstance> parentPathwayReviewedAuthoredRevisedInstances = parentPathway.getAttributeValuesList(ReactomeJavaConstants.reviewed);
+				parentPathwayReviewedAuthoredRevisedInstances.addAll(parentPathway.getAttributeValuesList(ReactomeJavaConstants.authored));
+				parentPathwayReviewedAuthoredRevisedInstances.addAll(parentPathway.getAttributeValuesList(ReactomeJavaConstants.revised));
+
+				for (GKInstance newIE : newInstanceEdits)
+				{
+					for (GKInstance parentPathwayIE : parentPathwayReviewedAuthoredRevisedInstances)
 					{
-						for (GKInstance pathwayIE : parentPathwayIEs)
-						{
-							if (rleIE.getDBID().equals(pathwayIE.getDBID()))
-							{
-								// The parent Pathway contained the IE.
-								hasRLEInstanceEdit = true;
-							}
-						}
-						if (hasRLEInstanceEdit)
+
+						// At time of writing, no 'existing' RlEs have a new IE in the immediate parent
+						// when using
+						// 'test_slice_20210118' as the current DB, and 'slice_previous_as_74' as the
+						// previous DB.
+						if (hasRLEInstanceEdit(newIE, parentPathwayIE))
 						{
-							// If it has the new IE, it might be not be the highest Pathway in the hierarchy
-							// with it.
-							// Continue looking up the hierarchy to try and find the HIGHEST Pathway with
-							// the IE.
-							GKInstance grandparentPathwayWithRLEInstanceEdit = getGrandparentPathwayWithRLEInstanceEdit(parentPathway, rleIE);
-							Map<GKInstance, GKInstance> reaction2newIE = new HashMap<>();
-							reaction2newIE.put(newRLE, rleIE);
-							// If 'Pathway' key exists, add to it
-							if (pathway2Reactions2NewIEs.get(grandparentPathwayWithRLEInstanceEdit) != null)
-							{
-								pathway2Reactions2NewIEs.get(grandparentPathwayWithRLEInstanceEdit).add(reaction2newIE);
-							}
-							else
-							{
-								// If 'Pathway' key doesn't exist, create it and then add to it.
-								Set<Map<GKInstance, GKInstance>> reaction2newIESet = new HashSet<>(Arrays.asList(reaction2newIE));
-								pathway2Reactions2NewIEs.put(grandparentPathwayWithRLEInstanceEdit, reaction2newIESet);
-							}
+							System.out.println("This IE exists in immediate parent Pathway");
```

Comment:
It might be good to add the InstanceEdit dbId or display name so we know which InstanceEdit exists.

Reply:
Also, if it exists in the `parentPathway` the if-block reports it here but it doesn't add it to the pathwayMap returned at the end of the method - does it need to be added or it's left out for a reason?

Reply:
Sure, though I've run the program numerous times and that has never been printed to the console. I'm not sure the condition actually will happen or not.